### PR TITLE
Add typescript and configure support for GUI

### PR DESCRIPTION
**Pull request by rupurt**
[#163414230]

### CODE REVIEW

**Comment by j16r** [comment]
Could we maybe add a single type annotation to our JS to see this working?

**Comment by rupurt** [comment]
@j16r this file doesn't deal with any types but I converted it to `tsx` and had to update the modules to relative import syntax. Does that suffice your need to see it working?

File: gui/src/App.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ class App extends PureComponent {
     }
   }
 
-  render () {
+  render (): JSX.Element {
```
Comment (rupurt) [inline gui/src/App.tsx]: @j16r return type

**Comment by NavyAdmiral** [comment]
Awesome! 🥇

**Pull request merged**